app/assets/v2/js/pages/fulfill_bounty.js

```diff
@@ -149,15 +149,53 @@ window.onload = function() {
                 }
 
                 var bountyId = result['standard_bounties_id'];
+                var bountyNetwork = result['network'];
 
-                bounty.fulfillBounty(
-                  bountyId,
-                  document.ipfsDataHash,
-                  {
-                    gasPrice: web3.toHex($('#gasPrice').val() * Math.pow(10, 9))
-                  },
-                  web3Callback
-                );
+                // before committing to transaction prompt,
+                // retrieve bounty owner's address since a
+                // creator cannot fulfill their own bounty.
+                var fromAddress = result['bounty_owner_address'];
+
+                if (fromAddress == account) {
+                  _alert({ message: 'The address that funded an issue cannot fulfill it.' });
+                  unloading_button($('.js-submit'));
+                  return;
+                }
+
+                browserNetworkIs(bountyNetwork, function(matchingNetworks) {
```

Comment:
hey @mbeacom @thelostone-mc @tra38 @KamesCG  you all are good with javascript... is there a way we can do multiple checks against the blockchain without all these async callbacks?  the nested callbacks make it really hard for me to follow whats going on here.

Reply:
We could do Promises / async-await but we'd need ES6 for that  😓